### PR TITLE
GENERIC_DOWNLOAD_PAGE #874

### PR DESCRIPTION
**Pull request by YashGovekar**
GENERIC_DOWNLOAD_PAGE #874 

Pushing first commit to check if everything works fine with the rebase :D

### CODE REVIEW

File: app/Providers/RouteServiceProvider.php [hunk 1]
```diff
@@ -207,6 +207,20 @@ private function mapAdminRoutes()
...
+            // downloads
+            Route::group([
+                'as'         => 'downloads.',
+                'prefix'     => 'downloads',
+                'middleware' => ['ability:admin, files'],
+            ], function () {
+                Route::get('/', 'DownloadController@index')->name('index');
+                Route::get('/create', 'DownloadController@create')->name('create');
+                Route::get('/edit/{id}', 'DownloadController@edit')->name('edit');
+                Route::post('/', 'FileController@store')->name('store');
```
Comment (nabeelio) [inline app/Providers/RouteServiceProvider.php:219]: These are going to break the existing FileController rules

File: app/Providers/RouteServiceProvider.php [hunk 2]
```diff
@@ -207,6 +207,20 @@ private function mapAdminRoutes()
...
 
+            // downloads
+            Route::group([
+                'as'         => 'downloads.',
+                'prefix'     => 'downloads',
+                'middleware' => ['ability:admin, files'],
+            ], function () {
+                Route::get('/', 'DownloadController@index')->name('index');
+                Route::get('/create', 'DownloadController@create')->name('create');
+                Route::get('/edit/{id}', 'DownloadController@edit')->name('edit');
```
Comment (nabeelio) [inline app/Providers/RouteServiceProvider.php:218]: Follow the Laravel conventions on these. I'm trying to wrap my head around how this is setup.

Comment (YashGovekar) [reply app/Providers/RouteServiceProvider.php:218]: Umm yeah, using it. Let me re-check the conventions.

File: app/Providers/RouteServiceProvider.php (hunk 1 above)
Comment (YashGovekar) [reply app/Providers/RouteServiceProvider.php:219]: Oh, so I copy thr `FileController `store to `Admin\DownloadController`?

Comment (nabeelio) [reply app/Providers/RouteServiceProvider.php:219]: FileController is already being used, moving it here into this namespace will break it

Comment (YashGovekar) [reply app/Providers/RouteServiceProvider.php:219]: Umm so you suggest that I should create a DownloadService and use it to store downloads?

Comment (nabeelio) [reply app/Providers/RouteServiceProvider.php:219]: No... I haven't entirely thought through this feature yet

Comment (YashGovekar) [reply app/Providers/RouteServiceProvider.php]: Oh, okay.
Let me assure you that for now it works fine.

**Comment by nabeelio** [comment]
I think we need to put this one on hold for the time being until I can decide what exactly to do and how to design this one. I haven't really thought of how I want this to look

**Comment by nabeelio** [comment]
@YashGovekar These are all the critical bugs right now, maybe comment on one you want to look at?
https://github.com/nabeelio/phpvms/issues?q=is%3Aopen+is%3Aissue+milestone%3A7.0.0+label%3Abug

And I will make clearer for this issue what to do do you can have that when you're ready

**Comment by YashGovekar** [comment]
> I think we need to put this one on hold for the time being until I can decide what exactly to do and how to design this one. I haven't really thought of how I want this to look

Oh okay, sure.

**Comment by YashGovekar** [comment]
> And I will make clearer for this issue what to do do you can have that when you're ready

Sure, I'll check them out and comment on the one I would like yo solve.

**Pull request closed without merging**